src/classes/UTIL_CurrencyCache.cls

```diff
@@ -77,6 +77,11 @@ public class UTIL_CurrencyCache implements UTIL_CurrencyCache.Interface_x {
             // If a set of rates was retrieved, find the effective rate based on the date
             if (cachedMapOfRatesByCurrencyCode.containsKey(currCode)) {
                 UTIL_CurrencyCache.CurrencyData currData = cachedMapOfRatesByCurrencyCode.get(currCode);
+                
+                //Currency data is null.  Return 1, to align with existing logic and avoid NPE.
```

Comment:
This comment is probably not entirely necessary.  It kinda just rehashes the code block below it.  Also, I don't know what NPE is.  Maybe something like this might explain it a little better:
> If the currency data can't be found in the cache, return a default rate of 1

Reply:
Daniel - after a review this morning, I'm going to lobby for the original solution of a simple null check.  

In this class, we have a couple calls to check the map keys like this : 
`if (!cachedMapOfRatesByCurrencyCode.containsKey(currCode))`

By not mapping null values, we are forcing that call to dig deeper, each check,  to see if currency data exists - which can lead to additional queries.  By mapping those nulls, we can tell future calls that we've already determined the value - so don't check again.

There's another method using this setup called getCurrDecimalPlaces, where another similar null check is already being made.

Thoughts?

Reply:
I did not notice that we have the same situation in `getCurrDecimalPlaces()` method and previously decided that caching a null CurrencyData is a valid scenario.  And you make a good argument for why:  Yes, the currency code may be invalid or just plain garbage, but at least we *know* that is the case, and don't have to recheck it.  I buy it.  Well done.  I'd still ditch the comment though.